### PR TITLE
Adding no-root-info state for a Log during Log-filtering.

### PR DESCRIPTION
**Pull request by Mercurrent**
Per init-strategy discussion.

### CODE REVIEW

File: loglist/logfilter.go [hunk 1]
```diff
@@ -45,11 +49,18 @@ func (ll *LogList) Compatible(rootedChain []*x509.Certificate, roots LogRoots) L
 		return compatible
 	}
 	for _, l := range ll.Logs {
-		// Check root is accepted.
-		for _, r := range roots[l.URL] {
-			if r.Equal(rootedChain[len(rootedChain)-1]) {
-				compatible.Logs = append(compatible.Logs, l)
-				break
+		// If root set is not set, we treat Log as compatible assuming no
+		// knowledge on its roots.
```
Comment (phad) [inline loglist/logfilter.go]: nit: (sorry) 'no knowledge of its roots' is probably what you mean?

File: loglist/logfilter_test.go [hunk 2]
```diff
@@ -70,7 +70,12 @@ func singleCert() []*x509.Certificate {
 
 func artificialRoots(source string) LogRoots {
 	rootCert, _ := x509util.CertificateFromPEM([]byte(source))
-	return LogRoots{"log.bob.io": []*x509.Certificate{rootCert}}
+	return LogRoots{
+		"log.bob.io": []*x509.Certificate{rootCert},
+		"ct.googleapis.com/racketeer/":[]*x509.Certificate{},
```
[the remark this answers is not in the document]
Comment (Mercurrent) [reply loglist/logfilter_test.go]: Wasn't me picking that name.

File: loglist/logfilter.go [hunk 3]
```diff
@@ -45,11 +49,18 @@ func (ll *LogList) Compatible(rootedChain []*x509.Certificate, roots LogRoots) L
...
+		if logRoots, ok := roots[l.URL]; !ok {
+			compatible.Logs = append(compatible.Logs, l)
+			continue
+		} else {
```
[the remark this answers is not in the document]
Comment (Mercurrent) [reply loglist/logfilter.go]: Done.

File: loglist/logfilter.go [hunk 4]
```diff
@@ -20,10 +20,12 @@ import (
...
 )
 
-// LogRoots maps Log-URLs (stated at LogList) to the list of their accepted root-certificates.
+// LogRoots maps Log-URLs (stated at LogList) to the list of their accepted
+// root-certificates.
 type LogRoots map[string][]*x509.Certificate
 
-// ActiveLogs creates a new LogList containing only non-disqualified non-frozen logs from the original.
+// ActiveLogs creates a new LogList containing only non-disqualified non-frozen
+//  logs from the original.
```
[the remark this answers is not in the document]
Comment (Mercurrent) [reply loglist/logfilter.go]: Fixed.

File: loglist/logfilter.go (hunk 1 above)
Comment (Mercurrent) [reply loglist/logfilter.go]: Fixed.

File: loglist/logfilter_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (phad) [reply loglist/logfilter_test.go]: Can you change it to something noticably different?  Or will that affect other areas of the tests?

I'm just cautious of having things that differ only by one character, it's easy for confusion to result.

File: loglist/logfilter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (phad) [reply loglist/logfilter.go]: Sorry, I am probably being a bit dumb with Github, but I can't see the change you've made to fix this!

[the remark this answers is not in the document]
Comment (phad) [reply loglist/logfilter.go]: Oh ignore that. I just can't see the previous version, but I can see that you fixed it.

[the remark this answers is not in the document]
Comment (phad) [reply loglist/logfilter.go]: Maybe this is because you force-pushed the latest commit?

File: loglist/logfilter.go [hunk 5]
```diff
@@ -50,7 +50,7 @@ func (ll *LogList) Compatible(rootedChain []*x509.Certificate, roots LogRoots) L
 	}
 	for _, l := range ll.Logs {
 		// If root set is not set, we treat Log as compatible assuming no
-		// knowledge on its roots.
+		// knowledge of its roots.
```
Comment (phad) [inline loglist/logfilter.go]: lgtm.

**Pull request merged**